### ISSUE TITLE
Cannot Use a Reserved Word as a Shorthand Property

### ISSUE THREAD

**Issue by clhiker** [swc-project/swc#2832]
### Describe the bug

SWC only checks keyword as shorthand property, but does not take into account the special cases as function parameters.

### Input code

```typescript
function foo({ await }) {}
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": false,
      "dynamicImport": false,
      "privateMethod": false,
      "functionBind": false,
      "exportDefaultFrom": false,
      "exportNamespaceFrom": false,
      "decorators": false,
      "decoratorsBeforeExport": false,
      "topLevelAwait": false,
      "importMeta": false
    },
    "transform": null,
    "target": "es5",
    "loose": false,
    "externalHelpers": false
  }
}
```

### Playground link

https://play.swc.rs/?version=1.2.111&code=H4sIAAAAAAAAA0srzUsuyczPU0jLz9eoVkgsT8wsUajVVKiuBQA7MQpUGgAAAA%3D%3D&config=H4sIAAAAAAAAA0VOOw6DMAy9i2fWLtygA4ewgkFBSRzZRmqKuHsTNcBkv4%2Ff8wGbOhgPyChK0jYtyfADI5CLqE58NhiqrVILBqVzAENZyZpFX1ULzEpdHSD65JfSkhzHLKT6SJjWcDnPGhR53htxgJVcZzuJnDatqWrind23fzjx%2FDQF%2FJYbJH4nI%2BH8vHl90h1ep15nstP5A1SlDpf7AAAA

### Expected behavior

It may be a solution, which parsing the function parameters before checking the shorthand properties.
```js
function foo(_ref) {
  var _await = _ref["await"];
}
```

### Version

1.2.111

### Additional context

I run swc use cmd as follows
```bash
npx swc --config-file=.swcrc poc.js -o poc.js
```
The environment is ubuntu 20.04 amd64.

**Comment by kwonoj** [comment]
This looks like same as babel's behavior, fyi: https://babeljs.io/repl/#?browsers=chrome%2054&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABMOcAUBvRBDA7tmKRAXwEpENig&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=true&targets=&version=7.16.4&externalPlugins=%40babel%2Fplugin-transform-modules-commonjs%407.16.0&assumptions=%7B%7D

**Comment by Austaras** [comment]
This is valid js though

**Comment by kwonoj** [comment]
I am not saying it's not a bug, just saying both shares same behavior.

**Comment by Austaras** [comment]
Oh, this is not a bug on the babel side. Shorthand property is still considered a identifier so they still confirm to reserved word rules, and `await` is a reserved word only in `module` context. Babel default source type to `module`, so it throws; If source type is set  to `script`, it will work fine.

I don't know if swc has such mechanism, and if we just assume everything is in es module, this issue should be closed.

**Comment by Austaras** [comment]
TL;DR: this code is only valid when `isModule` set to false

**Comment by kdy1** [comment]
Thank you!
